src/emit/instructions.rs

```diff
@@ -484,54 +486,58 @@ pub(super) fn process_instruction<'a, T: TargetRuntime<'a> + ?Sized>(
                 .build_call(bin.functions[cfg_no], &parms, "")
                 .unwrap()
                 .try_as_basic_value()
-                .left()
-                .unwrap();
+                .left();
```

Comment:
Why did you remove the unwrap from here if you are unwrapping the value everywhere else? Is there a case when this call is going to fail?

Reply:
Great question, thanks
In Soroban, there is no return codes for functions or a pointer to returned data.
Instead, the actual data is returned as is. so `ret` might be `None`.
Therefore, in this [line](https://github.com/salaheldinsoliman/solang/blob/84d5f7db6b5d3010b062a075b4c50ea43036f610/src/emit/instructions.rs#L538-L539)
`} else if let Some(value) = ret {
   w.vars.get_mut(&res[0]).unwrap().value = value;
 }`
`ret` is not always `unwrapped`
 
 Does that properly answer your question?